controllers/reconciliation/manager.go

```diff
@@ -62,16 +67,41 @@ func (r *ReconcileManager) MainReconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	// TODO before check entando-k8s-service app ready
+	cr := &v1alpha1.EntandoAppV2{}
 	if cr, err = r.reconcileComponent(ctx, req, "ComponentManager", r.reconcileComponentManager, images.FetchComponentManager()); err != nil {
 		return err
 	}
 
+	// =========================== start legacy section ===========================
 	// progress step not added because is not a business step but jsut technical
-	r.statusUpdater.SetReconcileProcessingComponent(ctx, req.NamespacedName, "Csv")
-	csvReconcile := legacy.NewLegacyReconcileManager(r.Client, r.Log)
-	if err = csvReconcile.ReconcileClusterServiceVersion(ctx, req, images); err != nil {
-		r.statusUpdater.SetReconcileFailed(ctx, req.NamespacedName, "CsvReconciliationFailed")
-		return err
+	if utils.IsOlmInstallation() {
+		r.statusUpdater.SetReconcileProcessingComponent(ctx, req.NamespacedName, "Csv")
+		legacyReconcile := legacy.NewLegacyReconcileManager(r.Client, r.Log)
+		if err = legacyReconcile.ReconcileClusterServiceVersion(ctx, req, images); err != nil {
+			r.statusUpdater.SetReconcileFailed(ctx, req.NamespacedName, "CsvReconciliationFailed")
+			return err
+		}
+	} else {
+		r.statusUpdater.SetReconcileProcessingComponent(ctx, req.NamespacedName, "ImageInfo")
+		legacyReconcile := legacy.NewLegacyReconcileManager(r.Client, r.Log)
```

Comment:
This is the same instruction as line 79? Could we move them outside of the if else?

Reply:
yep, i will make it common